pegasus/sites.v3/code.org/views/census_map.haml

```diff
@@ -0,0 +1,150 @@
+      },
+      styles: [{
+        where: "review CONTAINS 'approved'",
+        markerOptions: {
+          iconName: "ltblu_stars"
+        }
+      }, {
+        where: "review DOES NOT CONTAIN 'approved'",
+        markerOptions: {
+          iconName: "measle_brown"
+        }
+      }],
+    });
+
+    layer.setMap(gmap);
+
+    google.maps.event.addListener(layer, 'click', function(event) {
+      enableDrag();
+      event.infoWindowHtml = event.row['organization_name'].value;
+      event.infoWindowHtml += '<br>';
+    });
+    google.maps.event.addListener(gmap, 'zoom_changed', function(event) {
+      enableDrag();
+    });
+    google.maps.event.addListener(gmap, 'click', function(event) {
+      enableDrag();
+    });
+  }
+
+  function enableDrag() {
+    scrollwheelOption = true;
+    draggableOption = true;
+
+    var map_options = {
+      scrollwheel: scrollwheelOption,
+      draggable: draggableOption
+    }
+    gmap.setOptions(map_options);
+  }
+
+  function resizeMap() {
+    var map_width = $('#gmap').width();
+    var map_zoom = gmap.getZoom();
+    var map_center = gmap.getCenter();
+
+    // Max height of map is 2/3 of screen height so user can always scroll.
+    var max_height = $(window).innerHeight() * maxHeightPercentage;
+
+    var window_aspect_ratio = $(window).innerHeight() / $(window).innerWidth();
+
+    if  (window_aspect_ratio < 1) {
+      // Landscape window. Use the current 1:2 ratio map size.
+      map_height = map_width / 2;
+
+      // But, make sure this size doesn't exceed our max height.
+       map_height = Math.min(map_height, max_height);
+    }
+    else {
+      // Portrait window. Just use the max height so that it's taller and more usable.
+      map_height = max_height;
+    }
+
+    $('#gmap').height(Math.round(map_height));
+
+    if (map_width < 256 && map_zoom < 3) {
+      map_zoom = 0;
+      map_center = new google.maps.LatLng(22.2116,4.5775);
+    }
+    else if (map_width < 512 && map_zoom < 3) {
+      map_zoom = 1;
+      map_center = new google.maps.LatLng(22.2116,4.5775);
+    }
+    else if (map_zoom < 3) {
+      map_zoom = 2;
+      map_center = new google.maps.LatLng(22.2116,4.5775);
+    }
+
+    var map_options = {
+      center: map_center,
+      zoom: map_zoom,
+      minZoom: map_zoom,
+      scrollwheel: scrollwheelOption,
+      draggable: draggableOption
+    }
+
+    gmap.setOptions(map_options);
+  }
+
+  function addGeocomplete() {
+    var geocomplete_options = {
+    };
+
+    $("#geocomplete").geocomplete(geocomplete_options)
+      .bind("geocode:result", function(event, result){
+        gmap.fitBounds(result.geometry.viewport);
+      });
+
+    $("#btn-submit").click(function(){
+      $("#geocomplete").trigger("geocode");
+    });
+
+    $("#btn-reset").click(function(){
+      $('#geocomplete').val('');
+      gmap.setCenter(new google.maps.LatLng(22.2116,4.5775));
+      gmap.setZoom(2);
+      info_window.close();
+      resizeMap();
+    });
+  }
```

Comment:
All this JS should be in an entry point rather than in haml.

Reply:
Good point.  I need to merge this for bugbash tomorrow morning but would like to revisit this.  Even better would be to de-dupe this with the (virtually identical) JS for the hourofcode.com map. :/